analytic_engine/src/row_iter/merge.rs

```diff
@@ -83,6 +84,7 @@ define_result!(Error);
 #[derive(Debug)]
 pub struct MergeConfig<'a> {
     pub request_id: RequestId,
+    pub metrics_collector: Option<MetricsCollector>,
```

Comment:
When this collector will be None?

Reply:
When the caller doesn't need the metrics e.g. compaction.